staking/types/delegation.go

```diff
@@ -0,0 +1,391 @@
+}
+
+type delegationRespAlias DelegationResponse
+
+// MarshalJSON implements the json.Marshaler interface. This is so we can
+// achieve a flattened structure while embedding other types.
+func (d DelegationResponse) MarshalJSON() ([]byte, error) {
+	return json.Marshal((delegationRespAlias)(d))
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface. This is so we can
+// achieve a flattened structure while embedding other types.
+func (d *DelegationResponse) UnmarshalJSON(bz []byte) error {
+	return json.Unmarshal(bz, (*delegationRespAlias)(d))
+}
+
+// DelegationResponses is a collection of DelegationResp
+type DelegationResponses []DelegationResponse
+
+// String implements the Stringer interface for DelegationResponses.
+func (d DelegationResponses) String() (out string) {
+	for _, del := range d {
+		out += del.String() + "\n"
+	}
+	return strings.TrimSpace(out)
+}
+
+// RedelegationResponse is equivalent to a Redelegation except that its entries
+// contain a balance in addition to shares which is more suitable for client
+// responses.
+type RedelegationResponse struct {
+	Redelegation
+	Entries []RedelegationEntryResponse `json:"entries" yaml:"entries"` // nolint: structtag
+}
+
+// RedelegationEntryResponse is equivalent to a RedelegationEntry except that it
+// contains a balance in addition to shares which is more suitable for client
+// responses.
+type RedelegationEntryResponse struct {
+	RedelegationEntry
+	Balance Dec `json:"balance"`
+}
+
+// NewRedelegationResponse returns a new Redelegation Response
+func NewRedelegationResponse(d common.Address, vSrc, vDst common.Address, entries []RedelegationEntryResponse) RedelegationResponse {
+	return RedelegationResponse{
+		Redelegation{
+			DelegatorAddress:    d,
+			ValidatorSrcAddress: vSrc,
+			ValidatorDstAddress: vDst,
+		},
+		entries,
+	}
+}
+
+// NewRedelegationEntryResponse returns a new RedelegationEntryResponse
+func NewRedelegationEntryResponse(ch int64, ct time.Time, s Dec, ib, b Dec) RedelegationEntryResponse {
+	return RedelegationEntryResponse{NewRedelegationEntry(ch, ct, ib, s), b}
+}
+
+// String implements the Stringer interface for RedelegationResp.
+func (r RedelegationResponse) String() string {
+	out := fmt.Sprintf(`Redelegations between:
+  Delegator:                 %s
+  Source Validator:          %s
+  Destination Validator:     %s
+  Entries:
+`,
+		r.DelegatorAddress, r.ValidatorSrcAddress, r.ValidatorDstAddress,
+	)
+
+	for i, entry := range r.Entries {
+		out += fmt.Sprintf(`    Redelegation Entry #%d:
+      Creation height:           %v
+      Min time to unbond (unix): %v
+      Initial Balance:           %s
+      Shares:                    %s
+      Balance:                   %s
+`,
+			i, entry.CreationHeight, entry.CompletionTime, entry.InitialBalance, entry.SharesDst, entry.Balance,
+		)
+	}
+
+	return strings.TrimRight(out, "\n")
+}
+
+type redelegationRespAlias RedelegationResponse
```

Comment:
technically alias in go lang is of form: `type f = g`

Reply:
this is removed from latest commit